### ISSUE TITLE
AtLeastOneConstructor false-positives

### ISSUE THREAD

**Issue by dgroup** [dgroup/arch4u-pmd#129]
### Summary
| Key            |  Value |
|:---------------|:------|
| Rule           | AtLeastOneConstructor   |
| arch4u version       | 0.1.1|

### Your rule definition
```xml
<rule ref="category/java/codestyle.xml/AtLeastOneConstructor">
    <properties>
      <property name="violationSuppressXPath" value=".[ends-with(@SimpleName,'Test')]"/>
      <property name="ignoredAnnotations" value="lombok.Data|lombok.Value|lombok.Builder|lombok.NoArgsConstructor|lombok.RequiredArgsConstructor|lombok.AllArgsConstructor|org.springframework.context.annotation.Configuration|org.springframework.web.bind.annotation.ControllerAdvice|org.springframework.web.bind.annotation.RestController|lombok.experimental.SuperBuilder|org.springframework.stereotype.Component|org.springframework.stereotype.Service|org.springframework.stereotype.Repository|org.springframework.boot.context.properties.ConfigurationProperties|org.mapstruct.Mapper|javax.xml.bind.annotation.XmlRootElement|org.junit.jupiter.api.extension.ExtendWith|org.mockserver.junit.jupiter.MockServerSettings|org.springframework.test.context.ContextConfiguration|org.springframework.test.context.web.WebAppConfiguration|org.junit.runner.RunWith" />
    </properties>
  </rule>
```

### Describe the defect or false/positive rule violation
- [ ] Case 1: when class implement functional interface that doesn't need the constructor
```java
public class ToString implements java.util.Function<Object, String> {
     @Override
     public String apply(Object arg) {
         return arg.toString();
    }
}
```
- [x] Case 2: When we have unit test class, thus we don't need the constructor
```java
public class ClassTest {
     
     public void test() {
          // implementation
    }
}
```
- [ ] Case 3: When methods in class have some particular annotations like `@org.junit.Test`, etc.
```java
public class ClassTest {
     
     @org.junit.Test / @Before / @After
     public void test() {
          // implementation
    }
}
```
- [ ] Case 4
```java
public class ClassA implements JsonDeserializer {
    // implementation     
}
```
